### PR TITLE
[TASK] Address recent Coding Standards

### PR DESCRIPTION
**Pull request by garvinhicking**
See https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/4011

Note that the phpstorm configuration PNG image has been removed as it's quite outdated. The current list of code styles is vastly different in recent PhpStorm versions, so IMO it's better to reference it without an image?

### CODE REVIEW

**Review by linawolf — APPROVED** [review]

File: Documentation/Appendix/Cgl.rst [hunk 1]
```diff
@@ -19,8 +19,10 @@ This is just a very brief overview.
 PHP
 ===
 
-Make sure you are using the correct PHP codestyle. It is **PSR-1 / PSR-2** at
-the time of writing and specified in the :ref:`TYPO3 Coding Guidelines
+Make sure you are using the correct PHP codestyle. It is based on
+the `TYPO3 Coding Standards <https://github.com/TYPO3/coding-standards>`__,
+(which follows PHP-FIG, PSR-1/PSR-2 and PSR-12 at the time of this writing),
```
Comment (brotkrueml) [inline Documentation/Appendix/Cgl.rst:24]: Actually, it is PER-CS1.0 (which is identical to PSR-12) with quite some additions towards PER-CS2.0 in Core:
https://github.com/TYPO3/typo3/blob/main/Build/php-cs-fixer/config.php

File: Documentation/Appendix/IdePhpStormSetup.rst [hunk 2]
```diff
@@ -56,22 +56,20 @@ If you use the :file:`.editorconfig` file (which is included in the TYPO3 core),
 standard formatting rules are already setup automatically (e.g. indent with
 4 spaces for PHP files).
 
-However, the rules defined in :file:`.editorconfig` are very minimal, so at
-least set up PhpStorm to comply with PSR-1 / PSR-2 for PHP as described next.
+The rules defined in :file:`.editorconfig` are very minimal and
+it is suggested to see the :ref:`TYPO3 Coding Guidelines <t3coreapi:cgl>`
+for more information and toolchain configuration.
+
+However, you can use the following PhpStorm code style to comply with PSR-12, which
```
Comment (brotkrueml) [inline Documentation/Appendix/IdePhpStormSetup.rst:63]: As above, PER-CS1.0 with additions. AFAIK PhpStorm can be configured to use the php-cs-fixer configuration file (PHP > Quality > PHP CS Fixer, or similar).

File: Documentation/Appendix/IdePhpStormSetup.rst [hunk 3]
```diff
@@ -56,22 +56,20 @@ If you use the :file:`.editorconfig` file (which is included in the TYPO3 core),
...
 :guilabel:`File > Settings > Editor > Code Style > PHP > Set From > Predefined Style`
 
-Choose PSR-1 / PSR-2
-
-
-.. image:: /Images/External/Phpstorm/phpstorm-settings-codestyle-php-setfrom.png
-   :align: center
-   :alt: Setup PhpStorm Predefined Code Style
-
+Choose :guilabel:`PSR-12`.
```
Comment (brotkrueml) [inline Documentation/Appendix/IdePhpStormSetup.rst:72]: As above, the configuration of php-cs-fixer should be used directly.

File: Documentation/Setup/SetupIde.rst [hunk 4]
```diff
@@ -31,8 +31,9 @@ There is an .editorconfig file in the TYPO3 core repository.
 See http://EditorConfig.org for more information.
 
 Please note that the rules in :file:`.editorconfig` are very minimal. So, additionally,
-setup your Editor / IDE to use PSR-1 / PSR-2 for PHP files, and use the other
-standards as described in the :ref:`Coding Guidelines of TYPO3 <t3coreapi:cgl>`.
+setup your Editor / IDE to use PSR-12 (recommended at the time of this
```
Comment (brotkrueml) [inline Documentation/Setup/SetupIde.rst:34]: Like above, PERCS-1.0 with additions.

**Review by brotkrueml — CHANGES_REQUESTED** [review]

**Comment by garvinhicking** [comment]
@brotkrueml  I've rephrased this to try to better indicate the changing standards and be more abstract.

File: Documentation/Appendix/Cgl.rst [hunk 5]
```diff
@@ -19,9 +19,11 @@ This is just a very brief overview.
 PHP
 ===
 
-Make sure you are using the correct PHP codestyle. It is **PSR-1 / PSR-2** at
-the time of writing and specified in the :ref:`TYPO3 Coding Guidelines
-<t3coreapi:cgl>`.
+Make sure you are using the correct PHP codestyle. It is based on
```
Comment (brotkrueml) [inline Documentation/Appendix/Cgl.rst:22]:
```suggestion
Make sure you are using the correct PHP code style. It is based on
```

File: Documentation/Appendix/IdePhpStormSetup.rst [hunk 6]
```diff
@@ -56,22 +56,25 @@ If you use the :file:`.editorconfig` file (which is included in the TYPO3 core),
...
-least set up PhpStorm to comply with PSR-1 / PSR-2 for PHP as described next.
+The rules defined in :file:`.editorconfig` are very minimal and
+it is suggested to see the :ref:`TYPO3 Coding Guidelines <t3coreapi:cgl>`
+for more information and toolchain configuration. Please read 
+`PhpStorm documentation on using PHP CS Fixer <https://www.jetbrains.com/help/phpstorm/using-php-cs-fixer.html#installing-configuring-php-cs-fixer>`__
+for information on how to use the provided
+`config.php <https://github.com/TYPO3/typo3/blob/main/Build/php-cs-fixer/config.php>`__
+within PhpStorm.
+
+For a quick start, you can Howevern use the following PhpStorm code
```
Comment (brotkrueml) [inline Documentation/Appendix/IdePhpStormSetup.rst:67]:
```suggestion
For a quick start, you can use the following PhpStorm code
```

File: Documentation/Setup/SetupIde.rst [hunk 7]
```diff
@@ -31,8 +31,8 @@ There is an .editorconfig file in the TYPO3 core repository.
 See http://EditorConfig.org for more information.
 
 Please note that the rules in :file:`.editorconfig` are very minimal. So, additionally,
-setup your Editor / IDE to use PSR-1 / PSR-2 for PHP files, and use the other
-standards as described in the :ref:`Coding Guidelines of TYPO3 <t3coreapi:cgl>`.
+setup your Editor / IDE to use the recommendation as described in the
```
Comment (brotkrueml) [inline Documentation/Setup/SetupIde.rst:34]:
```suggestion
setup your editor / IDE to use the recommendation as described in the
```

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**